### ISSUE TITLE
Entity is currently unavailable

### ISSUE THREAD

**Issue by WittyNameHere** [alexander0042/pirate-weather-ha#8]
Has been this way for about 11 hours now. Weather from the API frontpage still seems to work. API key is still listed on my account and loaded in configuration.yaml. Thought it might be because I recently added a Pi-Hole filter (although more than 11 hours ago), but suspending it didn't work. I'm not sure where to find relevant errors in Home Assistant, as I mostly set it up and then left it to work, so if you would like some kind of log data just point me to where to find it.

**Comment by alexander0042** [comment]
That's a little unusual, but thanks for opening an issue, we'll figure it out!
I just pushed an update to the repo, so try upgrading that (either via HACS or git pull), and let me know if that fixes it? If it doesn't then I've clarified the logging to show up as PirateWeather, which should make it easier to see what's happening here

**Comment by WittyNameHere** [comment]
Hi, I'm afraid I'm a bit more confused now. All I have on my end is a Home Assistant installation with this in configuration.yaml:

```
weather:
  - platform: darksky
    api_key: [redacted]
```
I found out about this service through a Reddit post you made promoting it as an alternative to the missing Dark Sky resource. I don't have HACS installed or pull anything from a repo. Did the configuration steps change?

**Comment by alexander0042** [comment]
Ok, that helps! So if you installed this by copying the [pirateweather](https://github.com/alexander0042/pirate-weather-ha/tree/master/custom_components/pirateweather) folder from the repo and renaming it to darksky, then that's all that you should have to do. Are you seeing any errors about integration version in your logs, since I was getting that one earlier?

**Comment by WittyNameHere** [comment]
Re-acquired the files in that folder, but it appears this is due to something else. I'll remove the coordinates for privacy but I found the problem:

> Logger: homeassistant.components.darksky.weather
> Source: components/darksky/weather.py:273
> Integration: darksky (documentation, issues)
> First occurred: 9:00:55 AM (1 occurrences)
> Last logged: 9:00:55 AM
> 
> Unable to connect to Dark Sky. 403 Client Error: Forbidden for url: https://api.darksky.net/forecast/[string]/[coordinates]?units=us&lang=en

**Comment by WittyNameHere** [comment]
Update: So I think I've solved my issue. For whatever reason, I was trying to contact DarkSky instead of PA. The HASS logs now show that I'm going to your server, but I'm getting a similar 403 rate exceeded error. My configuration yaml was copied directly from your instructions, with a `scan_interval: '00:05:00'` line.

I'm not sure why HASS is exceeding rate limits. Do I need to ask to be unblocked or will it just go away after a while?

**Comment by alexander0042** [comment]
Ok, now we're getting somewhere (even if it's not quite where we want to be)! 

You're 100% correct about that scan interval- I meant for it to be set to every 15 minutes, since that's (roughly) the resolution of my back end, and seemed to strike a good balance between frequency and my AWS bill! I've corrected this in the config, so that's a great catch. 

However, even at 5 minutes, I'm a little surprised that you'd have hit the cap, since it's set (unofficially) to allow that many requests per month. Apologies if you've already done this, but can you double check that you've clicked subscribe on the [API page](https://pirateweather.net/admin/apis)? 

429 is kind of a weird error here, and I'm not totally sure what would be throwing it. If you send me an e-mail <api@alexanderrey.ca> with your e-mail address/ API key, I'd love to take a closer look!

**Comment by alexander0042** [comment]
Sorry for the slow reply here- I was camping this weekend, which wreaks havoc on my internet connectivity!

Start with the dull but necessary question, I wanted to check if you've restarted your HA core? It seems like it's not recognizing the custom Dark Sky module, which is sometimes fixed by just restarting it. Otherwise, are there any errors in the log after the restart, particularly with respect to an "integration manifest" or something?
